### PR TITLE
Add test for time zone detection

### PR DESCRIPTION
**Pull request by mattjohnsonpint**
Adds a test for time zone detection, per #551.

### CODE REVIEW

**Comment by mattjohnsonpint** [comment]
@kangax - Do you generate the test results from somewhere?  Or do you run the tests manually and fill in the results?

**Comment by webbedspace** [comment]
Results are run and entered into the `data-*.js` files manually (along with any footnotes necessary to explain any anomalies).

File: data-esintl.js [hunk 1]
```diff
@@ -912,6 +912,19 @@ exports.tests = [
         iojs1_1: false,
       },
     },
+    'resolvedOptions().timeZone defaults to the host environment': {
+      link: 'http://www.ecma-international.org/ecma-402/1.0/#sec-6.4',
+      exec: function(){/*
+        try {
```
Comment (webbedspace) [inline data-esintl.js:918]: This `try` block is not required - see README.md.

**Comment by mattjohnsonpint** [comment]
Sorry, but I'm confused still.  Do you have a test environment for all these different browsers and versions? Or do you expect me to add the test results to this PR?  I don't have access to a test bed with so many configurations already set up.

**Comment by kangax** [comment]
We don't have a dedicated test environment -- just a bunch of browsers locally which we check manually (at least that's how I do it, not sure about @webbedspace, @ljharb, and @zloirock)

I'm actually not sure how to deal with PR's without tests. Maybe we should still merge them and then fill in results at some point in the future?

**Comment by mattjohnsonpint** [comment]
I've not used them before, but I know there are services that provide this functionality.  [BrowserStack says it's free for OSS projects](https://www.browserstack.com/javascript-testing-api).  Maybe look into that?

**Pull request closed without merging**